challenges/generic-function-arguments/prompt.md

```diff
@@ -0,0 +1,80 @@
+## Why Functions Need Generics
+
+One of the most common needs a TypeScript developer faces is the need to pass generic type arguments to functions.  Then, those functions can use and manipulate those parameters.
+
+In many situations, we don't need functions to be generic.  The function might be written in such a way as to only work for one type of data (e.g. `string` or `number`).  But, other times, we write functions where we want to pass the data type along from input to output.
+
+## Understanding Generic Syntax
+
+In a previous challenge on [generic type arguments](todo-generic-type-arguments) we saw a `Row` type that looked like this:
+
+```ts
+interface Row<T> {
+  label: string;
+  value: T;
+  disabled: boolean;
+}
+```
+
+Imagine, though, that you had a function `createRow` that returns an object that looked like our `Row` type:
+
+```ts
+const createRow = (
+  label: string,
+
+  // how do we provide a type for `value`?
+  value: unknown,
+  
+  disabled = false,
+) => ({
+  label,
+  value,
+  disabled,
+});
+```
+
+Similarly to how we used the `<` and `>` for our `Row` interface, we place the `<` and `>` right before the parenthesis that start our function's arguments.
+
+```ts
+const createRow = <T>(
+  label: string,
+  value: T,
+  disabled = false,
+) => ({
+  //...
+})
+```
+
+This is great!  Now our `value` parameter has the right type!
+
+### A Word On A Syntax Quirk
+
+You may see the `<T,>` syntax out in the wild.  There's an unfortunate niggle with TypeScript's syntax that this is used to work around.  Imagine being a parser and seeing `const createRow = <T>`.  If you don't know what's coming next it could be ambiguous between these two things
+
+```ts
+// Thing 1: Start of JSX
+const createRow = <T>Some JSX stuff!</T>;
+
+// Thing 2: start of generic function
+const createRow = <T>(someArg: T) => {
+  /* some function stuff */
+}
+```
+
+Because of this ambiguity, we need to do something to _disambiguate_ these two situations to the TypeScript compiler.  In the early days it used to be common to do use `<T extends unknown>` (or `<T extends any>` in the days before `unknown` existed in TypeScript).  But over time people realized there's a shorter way: `<T,>`.  Because `,` is invalid in JSX, but valid in TypeScript to separate multiple arguments, it's sufficient to use this to disambiguate the two situations to the TypeScript compiler.
```

Comment:
```suggestion
Because of this ambiguity, we need to do something to _disambiguate_ these two situations to the TypeScript compiler.  In the early days it used to be common to use `<T extends unknown>` (or `<T extends any>` in the days before `unknown` existed in TypeScript).  But over time people realized there's a shorter way: `<T,>`.  Because `,` is invalid in JSX, but valid in TypeScript to separate multiple arguments, it's sufficient to use this to disambiguate the two situations to the TypeScript compiler.
```

Reply:
theres a lot of double spaces after periods. not a big deal but just noticing it